infra/gravity/opscenter.go

```diff
@@ -0,0 +1,164 @@
+// it is a minimal copy of only needed fields
+type ClusterSpecV2 struct {
+	// Status is a cluster status, initialized for existing clusters only
+	Status string `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+// parseClusterStatus will attempt to unmarshal the cluster status from tele get clusters output
+func parseClusterStatus(clusterName string, data []byte) (string, error) {
+	if len(data) == 0 {
+		return "", trace.BadParameter("missing cluster data")
+	}
+
+	if strings.Contains(string(data), fmt.Sprintf("cluster %v not found", clusterName)) {
+		return "", trace.NotFound("cluster not found")
+	}
+
+	cluster := ClusterV2{}
+	err := yaml.Unmarshal(data, &cluster)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	return cluster.Spec.Status, nil
+}
+
+// generateClusterConfig will generate a cluster configuration for the ops center based
+// on the built in template
+func generateClusterConfig(cfg ProvisionerConfig, clusterName string) (string, error) {
+	template, err := template.New("cluster").Parse(`
+kind: cluster
+version: v2
+metadata:
+  labels:
+    Name: {{ .ClusterName }}
+  name: {{ .ClusterName }}
+spec:
+  app: {{ .Cfg.Ops.App }}
+  aws:
+    keyName: ops
+    region: {{ .Cfg.Ops.EC2Region }}
+  nodes:
+  - profile: node
+    count: {{ .Cfg.NodeCount }}
+    instanceType: c4.2xlarge
+  provider: aws`)
+
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	buf := &bytes.Buffer{}
+	err = template.Execute(buf, struct {
+		Cfg         ProvisionerConfig
+		ClusterName string
+	}{
+		cfg,
+		clusterName,
+	})
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	return buf.String(), nil
+}
+
+// DestroyOpsFn will destroy the cluster by making a request to the ops center to de-provision the cluster
+func (c ProvisionerConfig) DestroyOpsFn(tc *TestContext, clusterName string) func() error {
+	return func() error {
+		log := tc.Logger().WithFields(logrus.Fields{
+			"cluster": clusterName,
+		})
+
+		// if the TestContext has an error, it means this robotest run failed
+		if tc.Context().Err() != nil {
+			log.WithError(tc.Context().Err()).Info("error caught in testing")
+
+			if !policy.DestroyOnFailure {
+				log.Info("skipped destroy due to error")
+				return trace.Wrap(tc.Context().Err())
+			}
+		} else {
+			if !policy.DestroyOnSuccess {
+				log.Info("skipped destroy on success due to policy")
+				return nil
+			}
+		}
+
+		log.Info("destroying cluster")
+
+		out, err := exec.Command("tele", "rm", "cluster", clusterName).CombinedOutput()
+		if err != nil {
+			return err
+		}
+		log.Debug("tele rm result: ", string(out))
+
+		// monitor the cluster until it's gone
+		timeout := time.After(DefaultTimeouts.Uninstall)
+		tick := time.Tick(5 * time.Second)
+
+		for {
+			select {
+			case <-timeout:
+				return trace.LimitExceeded("clusterDestroy timeout exceeded")
+			case <-tick:
+				// check provisioning status
+				status, err := getTeleClusterStatus(clusterName)
```

Comment:
any external commands need be executed with context.Context passed. it's no fun to debug robotest with tons of concurrent tests running and some external program got hung.